scripts/SREP/retry-pruning-cronjob/script.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Delete jobs that fail in openshift-sre-pruning namespace.
+
+set -x
+set -e
+set -o nounset
+set -o pipefail
+
+readonly PRUNING_NS=openshift-sre-pruning
+FailedJobs=()
+
+detect_job() {
+  echo "INFO: finding jobs producing error"
+  for current_ns in ${PRUNING_NS}; do
+    FailedJobs+=($(oc get jobs -n "$current_ns" -o=jsonpath='{.items[?(@.status.failed>0)].metadata.name}'))
+  done
+}
+
+delete_job() {
+  if [[ ${#FailedJobs[@]} == 0 ]]; then
+    echo "INFO: no failed jobs found exiting.."
+    return 1
+  else
+    echo "INFO: deleting jobs producing error"
+    for jobs in "${FailedJobs[@]}"; do
+      oc delete job "$jobs" -n "$PRUNING_NS"
+      FailedJobs=("${FailedJobs[@]:1}")
```

Comment:
Any reason you add this `FailedJobs=("${FailedJobs[@]:1}")`?  

The current logic of the `delete_job` function seems to be to delete each job in the FailedJobs array and then remove it from the array. 

However, `FailedJobs=("${FailedJobs[@]:1}")` is not necessary in this specific script, because the FailedJobs array is not used after the delete_job function. Once all jobs are deleted, the script exits.